ci/task/compile_models.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+# Usage:
+# ./compile_models.sh "${target}" ${hf_token}
+set -eo pipefail
+set -x
+
+TARGET=$1
+HF_TOKEN=$2
+
+if [[ ${TARGET} == cuda* ]]; then
+    conda run -n py38 pip install --pre -U -f https://mlc.ai/wheels mlc-ai-nightly-cu121
+elif [[ ${TARGET} == rocm* ]]; then
+    conda run -n py38 pip install --pre -U -f https://mlc.ai/wheels mlc-ai-nightly-rocm57
+else
+    conda run -n py38 pip install --pre -U -f https://mlc.ai/wheels mlc-ai-nightly
+fi
+
+conda run -n py38 pip install wheels/*.whl
+conda run -n py38 pip install huggingface_hub
+conda run -n py38 python ci/task/compile_models.py --targets "$TARGET" --hf-token "$HF_TOKEN"
```

Comment:
A principal we designed those bash scripts is that we want them to be run out of the box on user end if they have the environment configured properly. An example of such principal is:

```bash
# run locally in a user environment 
./ci/task/build_clean.sh # this will clean up the build arifacts
```

With this in mind, a script should not be implemented with the assumption that it is run in a conda environment with a specific name, in this case, "py38". Instead, this is moved to Jenkinsfile as part of the CI environment. See [this](https://github.com/mlc-ai/mlc-llm/blob/89867a0be2a066021ea2ed08d3c6b72934891bf7/ci/jenkinsfile.groovy#L133-L135) as an example.